### PR TITLE
docs: Add development environment section to readme

### PR DESCRIPTION
**Pull request by TerryHowe**
Each ORAS sub-project should have a section in the README.md about setting up a development environment, since each sub-project is unique.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,6 +11,27 @@
 Documentation for the ORAS CLI is located on
 the project website: [oras.land/cli](https://oras.land/cli/)
 
+## Development Environment Setup
+
+You will need `git`, `go` and `make` installed on your system to start development.
+Once you have the basic tools installed, fork the repository on github and clone it locally:
```
Comment (Mr-DG-Wick) [inline README.md:17]:
```suggestion
Once you have the basic tools installed, fork the repository on GitHub and clone it locally:
```

Minor typo fix suggestion : github could be replaced with GitHub

File: README.md [hunk 2]
```diff
@@ -11,6 +11,27 @@
...
+git clone git@github.com:${MY_GITHUB_NAME}/oras.git
+cd oras
+```
+To compile and run the tests, just run the `make` command.
+If you want to create an executable, run `make build` which will build executables for several different operating systems and architectures.
+The `uname -a` command should give you some hints on which one to use:
+```
+% uname -a
+Darwin 3c22fb4124 21.6.0 Darwin Kernel Version 21.6.0: Mon Dec 29 20:44:01 PST 2022; root:xnu-8020.240.18~2/RELEASE_X86_64 x86_64
+``
```
Comment (Mr-DG-Wick) [inline README.md:29]:
```suggestion

```
```
minor typo fix suggestion:   `` -> ```

**Review by Mr-DG-Wick — CHANGES_REQUESTED** [review]
Second this! Thanks for the PR @TerryHowe. I've given some minor suggestions that would help fix some unwanted behavior.

**Review by SteveLasker — APPROVED** [review]
Being more explicit on the details helps folks get started.
Thanks for the addition, @TerryHowe

File: README.md [hunk 3]
```diff
@@ -11,6 +11,26 @@
...
+MY_GITHUB_NAME="my-name"
+git clone git@github.com:${MY_GITHUB_NAME}/oras.git
+cd oras
+```
+To compile and run the tests, just run the `make` command.
+If you want to create an executable, run `make build` which will build executables for several different operating systems and architectures.
+The `uname -a` command should give you some hints on which one to use:
+```
+% uname -a
+Darwin 3c22fb4124 21.6.0 Darwin Kernel Version 21.6.0: Mon Dec 29 20:44:01 PST 2022; root:xnu-8020.240.18~2/RELEASE_X86_64 x86_64
```
Comment (Mr-DG-Wick) [inline README.md:28]:
```suggestion
```
% uname -a
Darwin 3c22fb4124 21.6.0 Darwin Kernel Version 21.6.0: Mon Dec 29 20:44:01 PST 2022; root:xnu-8020.240.18~2/RELEASE_X86_64 x86_64
```
```

Comment (Mr-DG-Wick) [inline README.md:28]: I think a closing ` ``` ` is missing after the line 28 that would close the code snippet :)

**Comment by Mr-DG-Wick** [comment]
LGTM 🚀

**Review by Mr-DG-Wick — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -11,6 +11,27 @@
...
+git clone git@github.com:${MY_GITHUB_NAME}/oras.git
+cd oras
+```
+To compile and run the tests, just run the `make` command.
+If you want to create an executable, run `make build` which will build executables for several different operating systems and architectures.
+The `uname -a` command should give you some hints on which one to use:
+```
+% uname -a
+```
+Darwin 3c22fb4124 21.6.0 Darwin Kernel Version 21.6.0: Mon Dec 29 20:44:01 PST 2022; root:xnu-8020.240.18~2/RELEASE_X86_64 x86_64
```
Comment (sajayantony) [inline README.md:29]: Do you want the output to into the code block. The readme currently renders without line breaks for Generally.

Comment (TerryHowe) [reply README.md:28]: fixed

**Review by shizhMSFT — COMMENTED** [review]
We should mention and probably merge this doc https://oras.land/cli/5_developer_guide/?

**Review by sajayantony — COMMENTED** [review]
Oh yes why don't we point to - https://oras.land/cli/5_developer_guide/

And add to that any improvements? 
Would avoid things going stale and having to keep them consistent in 2 places.

**Review by sajayantony — APPROVED** [review]

**Review by qweeah — APPROVED** [review]
LGTM. BTW @TerryHowe, the merging rule now checks for unresolved discussion. If you want to merge this PR, you need to **resolve all the comments.**

**Comment by TerryHowe** [comment]
Updated the docs here https://github.com/oras-project/oras-www/pull/119 and just linked to them in this PR. This was part of the CNCF template work to have a development guide in the readme.

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**